src/model/formula/Formula.PredicateAtom.js

```diff
@@ -75,6 +75,14 @@ class PredicateAtom extends Formula {
   setVariable(from, to){
     this.terms.forEach(term => term.setVariable(from, to));
   }
+
+  getSubFormulasCommitment(commitment){
+    return [];
+  }
+
+  getVariables(){
+    return this.terms.map(term => term.getVariables()).flat();
```

Comment:
[`flatMap(… => …)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) is more efficient than `map(… => …).flat()`